ArdusatSDK.cpp

```diff
@@ -98,211 +108,51 @@ void _writeBeginError(const char sensorName[] PROGMEM)
-  return start_sensor_or_err(ir_temperature_sensor_name, mlx90614_init);
-}
-
-void readInfraredTemperature(temperature_t & output) {
-  output.header.unit = DATA_UNIT_DEGREES_CELSIUS;
-  output.header.timestamp = millis();
-
-  output.header.sensor_id = SENSORID_MLX90614;
-  output.t = mlx90614_getTempCelsius();
-}
-
-/*
- * Luminosity
- */
-boolean beginLuminositySensor() {
-#if defined(TSL2591_LUMINOSITY)
-  return start_sensor_or_err(luminosity_sensor_name, tsl2591_init);
-#else
-  return start_sensor_or_err(luminosity_sensor_name, tsl2561_init);
-#endif
-}
-
-void readLuminosity(luminosity_t & output) {
-  output.header.unit = DATA_UNIT_LUX;
-  output.header.timestamp = millis();
-
-#if defined(TSL2591_LUMINOSITY)
-  output.header.sensor_id = SENSORID_TSL2591;
-  output.lux = tsl2591_getLux();
-#else
-  output.header.sensor_id = SENSORID_TSL2561;
-  output.lux = tsl2561_getLux();
-#endif
-}
-
-/*
- * UV Light
- */
-boolean beginUVLightSensor() {
-#if defined(SI1145_UV_LIGHT)
-  return start_sensor_or_err(uvlight_sensor_name, si1145_init);
-#else
-  return start_sensor_or_err(uvlight_sensor_name, ml8511_init);
-#endif
-}
-
-void readUVLight(uvlight_t & output, int pin) {
-  output.header.unit = DATA_UNIT_MILLIWATT_PER_CMSQUARED;
-  output.header.timestamp = millis();
-
-#if defined(SI1145_UV_LIGHT)
-  output.header.sensor_id = SENSORID_SI1145;
-  output.uvindex = si1145_getUVIndex();
-#else
-  output.header.sensor_id = SENSORID_ML8511;
-  output.uvindex = ml8511_getUV(pin);
-#endif
-}
-
-/*
- * Acceleration
- */
-boolean beginAccelerationSensor() {
-  return start_sensor_or_err(acceleration_sensor_name, lsm303_accel_init);
-}
-
-void readAcceleration(acceleration_t & output) {
-  output.header.unit = DATA_UNIT_METER_PER_SECONDSQUARED;
-  output.header.timestamp = millis();
-
-  output.header.sensor_id = SENSORID_ADAFRUIT9DOFIMU;
-  lsm303_getAccel(&(output.x), &(output.y), &(output.z));
-}
-
-/*
- * Magnetic Field
- */
-boolean beginMagneticSensor() {
-  return start_sensor_or_err(magnetic_sensor_name, lsm303_mag_init);
-}
-
-void readMagnetic(magnetic_t & output) {
-  output.header.unit = DATA_UNIT_MICROTESLA;
-  output.header.timestamp = millis();
-
-  output.header.sensor_id = SENSORID_ADAFRUIT9DOFIMU;
-  lsm303_getMag(&(output.x), &(output.y), &(output.z));
-}
-
-/*
- * Gyroscope
- */
-boolean beginGyroSensor() {
-  return start_sensor_or_err(gyro_sensor_name, l3gd20h_init);
-}
-
-void readGyro(gyro_t & output) {
-  output.header.unit = DATA_UNIT_RADIAN_PER_SECOND;
-  output.header.timestamp = millis();
-
-  output.header.sensor_id = SENSORID_ADAFRUIT9DOFIMU;
-  l3gd20h_getOrientation(&(output.x), &(output.y), &(output.z));
-}
-
-/*
- * Barometric Pressure
  */
-boolean beginBarometricPressureSensor() {
-  return start_sensor_or_err(pressure_sensor_name, bmp180_init);
-}
+void _writeUnavailableSensorError(const char sensorName[] PROGMEM, const char hardwareBuild[] PROGMEM) {
+  char err_msg[55];
```

Comment:
`unavailable_on_spaceboard_error_msg` is only 28 characters long, we don't need a 55 byte buffer here